src/components/implementation/no_interface/cFE_booter/osfiles.c

```diff
@@ -98,120 +106,169 @@ int32 OS_remove(const char *path)
 */
 int32 OS_rename(const char *old_filename, const char *new_filename)
 {
-    PANIC("Read Only Filesystem!");
-    return 0;
+    int32 ret = chk_path(old_filename);
+    if (ret != OS_FS_SUCCESS) return ret;
+    ret = chk_path(new_filename);
+    if (ret != OS_FS_SUCCESS) return ret;
+    return file_rename((char *)old_filename, (char *)new_filename);
 }
 
 /*
  * copies a single file from src to dest
 */
 int32 OS_cp(const char *src, const char *dest)
 {
-    PANIC("Read Only Filesystem!");
-    return 0;
+    char *src_path  = (char *)src;
+    char *dest_path = (char *)dest;
+
+    if (!src_path || !dest_path) return OS_FS_ERR_INVALID_POINTER;
+    if (strlen(src_path) > OS_MAX_PATH_LEN) return OS_FS_ERR_PATH_TOO_LONG;
+    if (strlen(dest_path) > OS_MAX_PATH_LEN) return OS_FS_ERR_PATH_TOO_LONG;
+    if (strlen(path_to_name(src_path)) > OS_MAX_FILE_NAME) return OS_FS_ERR_NAME_TOO_LONG;
+    if (strlen(path_to_name(dest_path)) > OS_MAX_FILE_NAME) return OS_FS_ERR_NAME_TOO_LONG;
+
+    int32 ret = chk_path(src_path);
+    if (ret != OS_FS_SUCCESS) return ret;
+    ret = chk_path_new(dest_path);
+    if (ret != OS_FS_SUCCESS) return ret;
+
+    return file_cp(src_path, dest_path);
 }
 
 /*
  * moves a single file from src to dest
 */
 int32 OS_mv(const char *src, const char *dest)
 {
-    PANIC("Read Only Filesystem!");
-    return 0;
+    char *src_path  = (char *)src;
+    char *dest_path = (char *)dest;
+
+    if (!src_path || !dest_path) return OS_FS_ERR_INVALID_POINTER;
+    if (strlen(src_path) > OS_MAX_PATH_LEN) return OS_FS_ERR_PATH_TOO_LONG;
+    if (strlen(dest_path) > OS_MAX_PATH_LEN) return OS_FS_ERR_PATH_TOO_LONG;
+    if (strlen(path_to_name(src_path)) > OS_MAX_FILE_NAME) return OS_FS_ERR_NAME_TOO_LONG;
+    if (strlen(path_to_name(dest_path)) > OS_MAX_FILE_NAME) return OS_FS_ERR_NAME_TOO_LONG;
+
+    int32 ret = chk_path(src);
+    if (ret != OS_FS_SUCCESS) return ret;
+    ret = chk_path_new(dest);
+    if (ret != OS_FS_SUCCESS) return ret;
+
+    return file_mv(src_path, dest_path);
 }
 
 /*
  * Copies the info of an open file to the structure
 */
 int32 OS_FDGetInfo(int32 filedes, OS_FDTableEntry *fd_prop)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    if (!fd_prop)                                   return OS_FS_ERR_INVALID_POINTER;
+    if (filedes <= 0 || filedes > MAX_NUM_FILES)    return OS_FS_ERR_INVALID_FD;
+
+    return file_FDGetInfo(filedes, fd_prop);
 }
 
 /*
 ** Check to see if a file is open
 */
 int32 OS_FileOpenCheck(char *Filename)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    int32 ret = chk_path(Filename);
+    if (ret != OS_FS_SUCCESS) return OS_FS_ERR_INVALID_POINTER;
+
+    struct fsobj *file = file_find(Filename);
+    if (!file) return OS_INVALID_POINTER;
+    if (!file->refcnt) return OS_FS_ERROR;
```

Comment:
if `refcnt == 0`, then you should have have access to an allocated file.  So this check either is superfluous, or shows a deeper problem about how you're using refcnts.